website/docs/basics/architecture-concepts.md

```diff
@@ -0,0 +1,248 @@
+---
+
+## Project structure
+
+
+[![docs-source](https://img.shields.io/badge/source-eightshift--libs-blue?style=for-the-badge&logo=php&labelColor=2a2a2a)](https://github.com/infinum/eightshift-libs)
+
+Eightshift Development Kit follows a strict project structure to support autoloading, dependency injection and `wp boilerplate` commands, as well as conventions to increase codebase understanding between developers. As an added benefit, following this structure and conventions makes your codebase more consistent and organized.
+
+While Eightshift Development Kit provides required theme files and common theme files such as `style.css` and `functions.php`, they're used differently to what you might expect if you've worked with WordPress themes previously. For instance, `style.css` is used only to register theme metadata, and doesn't contain any actual styles, as they're built using Webpack and use cache busting. `functions.php` simply bootstraps the execution lifecycle. We also provide very basic template partials, such as `single.php`, which you can acommodate to your needs. An idea to consider is replacing some of them with pages that are editable in Gutenberg.
+
+While you'll find some additional configuration files such as `webpack.config.js` in the project root, the majority of your projects codebase should live in the `src` directory, which is autoloaded via Composer following PSR-4 conventions. This implies that every directory under `src` is a namespace under your base namespace (which is your project name), and PHP files inside of those directories are classes in that namespace, with exceptions for filenames that aren't in `StudlyCase`. 
+
+An Eightshift Development Kit convention is to structure namespaces around features and functionality, so for instance, all custom post types would be part of the `ProjectName\CustomPostType` namespace.
+
+## Object-oriented paradigms
+
+Eightshift Libs is entirely object-oriented, providing abstract classes for internal library functionality, interfaces you can use to implement certain classes to make use of Eightshift Development Kit features, as well as concrete class implementations for use in projects.
+
+Our use of OOP allows you to extend, modify or even replace Eightshift Development Kit functionality safely, without breaking compatibility or harming your project.
+
+One of the most important interfaces in Eightshift Libs is the `ServiceInterface`. Classes which implement it are Service. This isn't a common pattern in PHP OOP. In Eightshift Development Kit, a Service class represents a part of the project's functionality, and has a `register() : void` method that sets up the action and filter hooks for the Service. By convention, it also contains a method which provided as a callback to the registered hooks. If you need to add an action or a filter, you should do that in a Service. All your Service classes are automatically loaded, and can be injected using dependency injection.
```

Comment:
```suggestion
One of the most important interfaces in Eightshift Libs is the `ServiceInterface`. Classes that implement it are called service classes. This isn't a common pattern in PHP OOP. In Eightshift Development Kit, a service class represents a part of the project's functionality and has a `register` method that sets up the action and filter hooks for the service. If you need to add an action or a filter, you should do that in a service class. All your service classes are automatically loaded and can have dependencies injected using dependency injection.
```

I removed the 

> By convention, it also contains a method which provided as a callback to the registered hooks.

Because I'm not 100% sure what you meant by this sentence 🤷🏼‍♂️

Reply:
That's entirely fair. I meant to say that the service class is usually the same class that provides the callback, i.e. that the register method adds hooks to `[$this, 'someMethod']`, but that's not too relevant and can be skipped.